examples/memZono/SLAM.m

```diff
@@ -48,39 +47,34 @@
 
 
 %% simulation
-
+xDims = @(k) memZono.genKeys(sprintf('x_%ie',k),1:2);
+vKeys = @(k) sprintf('v_%ie',k);
+lDims = @(i) sprintf('L_%ie',i);
+lKeys = @(k,i) ['L_',num2str(k),'_',num2str(i)];
+lCons = @(k,i) sprintf('L%i_k%i',i,k);
 for k = 1:T     % loop over every time step 
 
     fprintf('--- k = %i -----------------\n',k-1)   % print time step
 
-    if k > 1                                                        % not initial condition
-        % u(:,k-1) = random_sample_zonotope(U);                     % determine actual input
-        mV = memZono(V,sprintf('v_%ie',k-1));                       % memZono version of the system uncertainty
-        newDimKeys = memZono.genKeys(sprintf('x_%ie',k-1),1:2);                          % create dimension keys for the new time step
-        X_nom{k} = X_nom{k-1}.transform([],F,{},newDimKeys) + mV.transform([],eye(2),{},newDimKeys);  % update state set factors
-        %X_nom{k}.c = X_nom{k}.c + G*u(:,k);                         % update state set center
-        x(:,k) = F*x(:,k-1) + G*u(:,k) + random_sample_zonotope(V); % determine actual state value
-        X{k} = [X{k-1}; X_nom{k}];                                  % insert next time step to state-landmark zonotope
+    if k > 1 % not initial condition
+        mV = memZono(V,xDims(k),vKeys(k-1)); % memZono version of the system uncertainty
+        X_nom{k} = X_nom{k-1}.map(F,xDims(k-1),xDims(k)) + mV; % determine actual state value (X_nom_{k} = F X_nom_{k-1} \oplus V)
+        X{k} = [X{k-1}; X_nom{k}]; % insert next time step to state-landmark zonotope
+        
+        x(:,k) = F*x(:,k-1) + G*u(:,k-1) + random_sample_zonotope(V); % determine actual state value
     end
 
     for i=1:num_l  % loop over each landmark
-
         r = landmarks{i}-x(:,k);    % actual vector displacement between vehicle and landmark
         theta = atan2(r(2),r(1));   % angle made relative to vehicle heading
 
-        if norm(r) <= landmark_radius               % vehicle can see landmark
-
+        if norm(r) <= landmark_radius  % vehicle can see landmark
             fprintf('k = %i --> Landmark %i\n',k-1,i);  % print time-step and landmark measured
             RH = rotate_zonotope(H,theta);              % rotated measurement noise zonotope
             r_m{k,i} = r + random_sample_zonotope(RH);  % noisy measurement of landmark
```

Comment:
```suggestion
            r_m{k,i} = r + random_sample_zonotope(RH);  % noisy measurement of landmark
            % this could be a suggestion... such as your comment from before changing the lKeys(k,i)
            L{k,i} = relabelDims(X_nom{k},xDims(k),lDims(i)) ... L_{k,i} = X_nom_{k} \oplus measuremnt...
                + memZono(RH + r_m{k,i},lDims(i),lKeys(k,i)); % calculate landmark position from actual position and distance measurement (r_m and RH) ... L{k,i} = 
            X{k} = X{k}.and(L{k,i}, lCons(k,i));% add new landmark to X{k}
```

Reply:
already fixed...